--- wolfssl-5.7.6-stable/wolfssl/wolfcrypt/types.h.orig	2024-12-31 18:58:22.000000000 +0100
+++ wolfssl-5.7.6-stable/wolfssl/wolfcrypt/types.h	2025-03-30 13:27:06.041537066 +0200
@@ -229,7 +229,7 @@ decouple library dependencies with stand
         #if !defined(SIZEOF_LONG_LONG) && !defined(SIZEOF_LONG)
             #if (defined(__alpha__) || defined(__ia64__) || \
                 defined(_ARCH_PPC64) || defined(__ppc64__) || \
-                defined(__x86_64__)  || defined(__s390x__ ) || \
+                (defined(__x86_64__) && defined(__LP64__)  || defined(__s390x__ ) || \
                 ((defined(sun) || defined(__sun)) && \
                  (defined(LP64) || defined(_LP64))) || \
                 (defined(__riscv_xlen) && (__riscv_xlen == 64)) || \
@@ -300,7 +300,7 @@ decouple library dependencies with stand
         (defined(__mips64) && \
          ((defined(_ABI64) && (_MIPS_SIM == _ABI64)) || \
           (defined(_ABIO64) && (_MIPS_SIM == _ABIO64)))) || \
-         defined(__x86_64__) || defined(_M_X64)) || \
+         (defined(__x86_64__) && defined(__LP64__)) || defined(_M_X64)) || \
          defined(__aarch64__) || defined(__sparc64__) || defined(__s390x__ ) || \
         (defined(__riscv_xlen) && (__riscv_xlen == 64)) || defined(_M_ARM64) || \
         defined(__aarch64__) || defined(__ppc64__) || \
--- wolfssl-5.6.3-stable/wolfcrypt/src/siphash.c.orig	2023-06-19 06:16:56.000000000 +0200
+++ wolfssl-5.6.3-stable/wolfcrypt/src/siphash.c	2023-09-11 19:38:45.445322685 +0200
@@ -353,7 +353,7 @@ int wc_SipHashFinal(SipHash* sipHash, un
     return ret;
 }
 
-#if !defined(WOLFSSL_NO_ASM) && defined(__GNUC__) && defined(__x86_64__) && \
+#if !defined(WOLFSSL_NO_ASM) && defined(__GNUC__) && defined(__x86_64__) && defined(__LP64__) && \
     (WOLFSSL_SIPHASH_CROUNDS == 1 || WOLFSSL_SIPHASH_CROUNDS == 2) && \
     (WOLFSSL_SIPHASH_DROUNDS == 2 || WOLFSSL_SIPHASH_DROUNDS == 4)
 
